test/server/cards/03_TWI/upgrades/OldAccessCodes.spec.ts

```diff
@@ -0,0 +1,50 @@
+describe('Old Access Codes', function() {
+    integration(function(contextRef) {
+        describe('Old Access Codes ability', function() {
+            it('should draw a card', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['old-access-codes'],
+                        groundArena: ['battlefield-marine'],
+                        deck: ['ahsoka-tano#always-ready-for-trouble']
+                    },
+                    player2: {
+                        groundArena: ['r2d2#ignoring-protocol', 'atst']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.oldAccessCodes);
+                context.player1.clickCard(context.battlefieldMarine);
+
+                expect(context.player1.handSize).toBe(1);
+                expect(context.ahsokaTano).toBeInZone('hand');
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('should not draw a card', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['old-access-codes'],
+                        groundArena: ['battlefield-marine']
+                    },
+                    player2: {
+                        groundArena: ['r2d2#ignoring-protocol']
```

Comment:
Do you mind putting an upgrade on R2 just to make sure that isn't getting counted, our arena methods have had some issues in the past. So like `groundArena: [{ card: 'r2d2#ignoring-protocol', upgrades: ['academy-training'] }]` is good